bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java

```diff
@@ -0,0 +1,181 @@
+    return result.isEmpty();
+  }
+
+  @Override
+  public ByteString getKey(Result result) {
+    return ByteStringer.wrap(result.getRow());
+  }
+
+  public static class ResultRowBuilder implements RowBuilder<Result> {
+    private byte[] currentKey;
+    private String family;
+    private ByteString qualifier;
+    private List<String> labels;
+    private long timestamp;
+    private ByteString value;
+
+    private Map<String, List<RowCell>> cells = new TreeMap<>();
+    private List<RowCell> currentFamilyCells = null;
+    private String previousFamily;
+    private int totalCellCount = 0;
+
+    @Override
+    public void startRow(ByteString rowKey) {
+      this.currentKey = ByteStringer.extract(rowKey);
+    }
+
+    @Override
+    public void startCell(
+        String family, ByteString qualifier, long timestamp, List<String> labels, long size) {
+      this.family = family;
+      this.qualifier = qualifier;
+      this.timestamp = timestamp;
+      this.labels = labels;
+      this.value = ByteString.EMPTY;
+    }
+
+    @Override
+    public void cellValue(ByteString value) {
+      this.value = this.value.concat(value);
+    }
+
+    /**
+     * Adds a Cell to {@link RowCell}'s map which is ordered by family. cells received from {@link
+     * RowBuilder} has ordering as:
+     *
+     * <ul>
+     *   <li>family names clustered, but not sorted
+     *   <li>qualifiers in each family cluster is sorted lexicographically
+     *   <li>then descending by timestamp
+     * </ul>
+     *
+     * The end result will be that {@link RowCell} are ordered as:
+     *
+     * <ul>
+     *   <li>lexicographical by family
+     *   <li>then lexicographical by qualifier
+     *   <li>then descending by timestamp
+     * </ul>
+     *
+     * A flattened version of the {@link RowCell} map will be sorted correctly.
+     */
+    @Override
+    public void finishCell() {
+      if (!Objects.equals(this.family, this.previousFamily)) {
+        previousFamily = this.family;
+        currentFamilyCells = new ArrayList<>();
+        cells.put(this.family, this.currentFamilyCells);
+      }
+
+      RowCell rowCell =
+          new RowCell(
+              this.currentKey,
+              this.family.getBytes(),
+              ByteStringer.extract(this.qualifier),
+              TimestampConverter.bigtable2hbase(this.timestamp),
+              ByteStringer.extract(this.value),
+              this.labels);
+      this.currentFamilyCells.add(rowCell);
+      totalCellCount++;
+    }
+
+    /**
+     * This method flattens the {@code cells} which has a map of Lists keyed by family name. The
+     * {@code cells} treeMap is sorted lexicographically, and each List is sorted by qualifier in
+     * lexicographically ascending order, and timestamp in descending order.
+     *
+     * @return an object of HBase {@link Result}.
+     */
+    @Override
+    public Result finishRow() {
+      if (totalCellCount == 0) {
```

Comment:
Did you check that Row has been started before?
Did you check that there is no 'started' cells?